pkgs/development/interpreters/perl/default.nix

```diff
@@ -133,7 +133,9 @@ let
       sha256 = "072j491rpz2qx2sngbg4flqh4lx5865zyql7b9lqm6s1kknjdrh8";
     };
 
-    nativeBuildInputs = [ buildPackages.stdenv.cc ];
+    # Hacky! But not sure how else we can access a native-targeted gcc6
+    # https://github.com/arsv/perl-cross/issues/60
+    nativeBuildInputs = [ buildPackages.buildPackages.gcc6 ];
```

Comment:
@Ericson2314 can we do better than this?

Reply:
`depsBuildBuild = [ buildPackages.stdenv.cc ];` I think, though I'm still not sure in cross stuff.

Reply:
This is at least what we do in (some) other cases when a build needs to compile stuff to use it during the build already.

Reply:
@vcunat That's what was here, but I specifically need gcc6

Reply:
I didn't see `depsBuildBuild` here.

Reply:
Ah, right, it was nativeBuildInputs. But even putting in depsBuildBuild, simple `buildPackages.gcc6` gives me a cross-compiler, I need the double `buildPackages` (which seems wrong).

Reply:
That's surprising to me.  It might be a bug in the implementation, docs, or my understanding...

Reply:
https://github.com/NixOS/nixpkgs/issues/35543

Reply:
OK #35543 should now explain everything, but the short answer is this should be
```
depsBuildBuild = [ gcc6 ];
```
(or, without relying on `callPackage` splicing
```
depsBuildBuild = [ buildPackages.buildPackages.gcc6 ];
```
)